src/devices/src/virtio/net/device.rs

```diff
@@ -520,6 +520,11 @@ impl Net {
 
     pub fn process_rx_queue_event(&mut self) {
         METRICS.net.rx_queue_event_count.inc();
+        if !self.queues[RX_INDEX].ready {
```

Comment:
These checks are not enough. The `ready` flag on a queue does not involve that the queue is also fully configured and valid. See `queue.is_valid()` function.
Furthermore even if a particular queue is valid, the device might still not be fully activated.

Ideally we should check that the `device_activated` field from `struct MmioTransport` has been set.

Another option that doesn't need us to get a handle on the `MmioTransport` object, would be to verify that the `trait VirtioDevice::activate()` function implemented by the `NetDevice` object was called and returned success.

Reply:
Addressed this issue with an approach which enforces the device to implement `is_activated` and `set_device_activated` methods, bound to `VirtioDevice` trait.